website/docs/components/form/primitives/partials/guidelines/guidelines.md

```diff
@@ -9,6 +9,16 @@
 
 - Use helper text to give the user extra details about the data you’re asking them to input, e.g., formatting requirements such as MM-DD-YYYY.
 
+### Form::CharacterCount
+
+Use a character count to communicate the maximum number of characters a field can accept before reaching an invalid state. By default, the character count displays the number of characters entered by the end user, compared to the maximum allowed length of the input. More custom implementations are supported via the exposed `currentLength`, `maxLength`, and `remainingLength` hooks. 
+
+<!-- TODO: link to code documentation -->
+
+![character count examples within a text area](/assets/components/form/primitives/primitives-example-textarea.png)
+
+![character count examples within a text input](/assets/components/form/primitives/primitives-example-textfield.png)
```

Comment:
Dropping a reminder that this section will need to be slightly adjusted after adding the change to include a minLength property.